impl/sql/keysets/keysets.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package keysets implements the storage.KeySets interface.
+package keysets
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/google/keytransparency/core/storage"
+
+	"github.com/golang/protobuf/proto"
+
+	tpb "github.com/google/keytransparency/core/api/type/type_proto"
+)
+
+const (
+	schema = `
+CREATE TABLE IF NOT EXISTS KeySets(
+	InstanceID            BIGINT NOT NULL,
+        DomainID              VARCHAR(40) NOT NULL,
+        AppID                 VARCHAR(40) NOT NULL,
+        KeySet                MEDIUMBLOB NOT NULL,
+        PRIMARY KEY(InstanceID,DomainID,AppID)
+);`
+
+	getSQL = `SELECT * FROM KeySets WHERE InstanceID = ? AND DomainID = ? AND AppID = ?`
+	setSQL = `INSERT INTO KeySets (InstanceID, DomainID, AppID, KeySet) VALUES (?, ?, ?, ?);`
+)
+
+// Storage stores keysets.
+type Storage struct {
+	db *sql.DB
+}
+
+type keyset struct {
+	InstanceID int64
+	DomainID   string
+	AppID      string
+	KeySet     []byte
+}
+
+// newkeyset converts a tpb.KeySet to keyset.
+func newkeyset(instance int64, domainID, appID string, k *tpb.KeySet) (*keyset, error) {
```

Comment:
Can this be `newKeyset` or must the case match the returned type exactly?

Reply:
renamed